### DISCUSSION TITLE
Override `GitInfo` (Gradle)

### DISCUSSION THREAD

**Discussion by joschi** [quarkusio/quarkus#44311] [Q&A]
Hey folks, we are using a build process in which the `.git/` directory of the repository is not available at build-time making [`InfoProcessor#gitInfo()`](https://github.com/quarkusio/quarkus/blob/main/extensions/info/deployment/src/main/java/io/quarkus/info/deployment/InfoProcessor.java#L55-L131) a no-op, so that [`GitInfo`](https://github.com/quarkusio/quarkus/blob/3.16.1/extensions/info/runtime/src/main/java/io/quarkus/info/GitInfo.java) won't be available during runtime.

Is there a way to override this information either with a custom properties file (as created by [`gradle-git-properties`](https://github.com/n0mer/gradle-git-properties)) or via system properties or environment variables?

[a comment by an automated account is not shown]

**Comment by gsmet** [comment]
This is an interesting case.
I'm a bit torn between telling you to open a feature request as your use case makes a lot of sense and telling you that the `GitInfo` part is really about extracting things from Git and that you can create a small extension that creates a `InfoBuildTimeContributorBuildItem` to include the info you will extract from wherever you want to get the info from.

I think the latter is probably the best and most flexible solution for you. You'll see that creating an extension is quite easy.

See https://quarkus.io/guides/building-my-first-extension for an intro and it should be as simple as having a dependency on `quarkus-info` and producing the build item.

Feel free to ask more questions here if you need additional guidance.
